prototypes/buildings/arthurian-pen.lua

```diff
@@ -542,43 +68,50 @@ RECIPE {
     }
 }:add_unlock("land-animals-mk04")
 
+local FULL_CRAFTING_SPEED = 1 -- crafting speed when full of mk01 modules
+local MODULE_SLOTS = 8
+local NAME = "arthurian-pen-mk0"
+local SUBGROUP = "py-alienlife-farm-buildings-mk0"
+local ENERGY_USAGE = {"200kW","300kW", "500kW", "700kW"}
+
+for i = 1, 4 do
 ITEM {
     type = "item",
-    name = "arthurian-pen-mk04",
-    icon = "__pyalienlifegraphics__/graphics/icons/arthurian-pen-mk04.png",
+    name = NAME .. i,
```

Comment:
Using locals within the loop for common references like `NAME .. i` would make these blocks more readable.